libvast/src/system/index.cpp

```diff
@@ -707,21 +707,9 @@ void schedule_lookups(index_state& st) {
     if (!partition_actor) {
       // We need to mark failed partitions as completed to avoid clients going
       // out of sync.
-      for (auto qid : next->queries) {
-        auto it = st.pending_queries.queries().find(qid);
-        if (it == st.pending_queries.queries().end()) {
-          VAST_WARN("{} tried to access non-existant query {}", *st.self, qid);
-          continue;
-        }
-        auto& query_state = it->second;
-        query_state.completed_partitions++;
-        if (query_state.completed_partitions
-            == query_state.requested_partitions)
-          st.self->send(query_state.client, atom::done_v);
-        if (query_state.completed_partitions
-            == query_state.candidate_partitions)
-          st.pending_queries.queries().erase(qid);
-      }
+      for (auto qid : next->queries)
+        if (auto client = st.pending_queries.handle_completion(qid))
+          st.self->send(*client, atom::done_v);
```

Comment:
That would indeed be great, but the current index client protocol doesn't allow it. I already had a handler to do this but removed it again in e616c1326f4fa78a71322ef876c5f9968014edab because it wasn't necessary for this PR.